### PR TITLE
GraphQl shipping address coverage

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description
https://github.com/magento/graphql-ce/issues/62
Added concept for shipping address coverage

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=192) <br/>All committers have signed the CLA.

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(ContextInterface $context, CartInterface $cart, array $shippingAddresses): void
+    {
+        if (count($shippingAddresses) > 1) {
+            throw new GraphQlInputException(
+                __('Multiple addresses do not allowed here!')
```
Comment (keharper) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php]: Change exception text to 

You cannot specify multiple shipping addresses.

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            );
+        }
+        $shippingAddress = current($shippingAddresses);
+        $customerAddressId = $shippingAddress['customer_address_id'] ?? null;
+        $addressInput = $shippingAddress['address'] ?? null;
+
+        if (!$customerAddressId && !$addressInput) {
+            throw new GraphQlInputException(
+                __('Shipping address should contain either "customer_address_id" or "address" input.')
```
Comment (keharper) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php]: Change exception text to

The shipping address must contain either "customer_address_id" or "address".

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            );
+        }
+        if ($customerAddressId && $addressInput) {
+            throw new GraphQlInputException(
+                __('Shipping address can\'t contain "customer_address_id" and "address" input at the same time.')
```
Comment (keharper) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php]: Change exception text to 

The shipping address cannot contain "customer_address_id" and "address" at the same time.

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            );
+        }
+        if ($customerAddressId) {
+            if ((!$context->getUserId()) || $context->getUserType() == UserContextInterface::USER_TYPE_GUEST) {
+                throw new GraphQlAuthorizationException(
+                    __(
+                        'Address management allowed only for authorized customers.'
```
Comment (keharper) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php]: Change exception text to

Guest users cannot manage addresses.

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressesOnCartInterface.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+namespace Magento\QuoteGraphQl\Model\Cart;
+
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Quote\Api\Data\CartInterface;
+
+/**
+ * Extension point for setting shipping addresses for a specified shopping cart
+ *
+ * All objects that are responsible for set shipping addresses on cart via GraphQl should implement this interface.
```
Comment (keharper) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressesOnCartInterface.php]: Change "set" to "setting" and insert "a" between "on" and "cart"

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 6]
```diff
@@ -6,17 +6,24 @@ type Query {
 }
 
 type Mutation {
+    createEmptyCart: String @resolver(class: "\\Magento\\QuoteGraphQl\\Model\\Resolver\\Cart\\CreateEmptyCart") @doc(description:"Creates empty shopping cart for guest or logged in user")
```
Comment (keharper) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]: Change description to

Creates an empty shopping cart for a guest or logged in user

**Review by keharper — CHANGES_REQUESTED** [review]

**Comment by VitaliyBoyko** [comment]
Hi @keharper 
Thank you for review.
I've fixed all the issues you've mentioned.
Please check :)

**Review by keharper — APPROVED** [review]

**Pull request merged**